master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,107 @@
+
+We recommend creating default deny using **Calico global network policy**, which is non-namespaced. It applies to both workloads (VMs and containers) and hosts (computers that run the hypervisor for VMs, or container runtime for containers). This supports a conservative security stance towards protecting resources.
+
+#### Calico network policy: order matters
+
+When you create Calico network policies to allow traffic, you can assign an **order** number that indicates the order you want policy applied relative to other policies. To ensure that your Calico default deny policy remains in effect at all times, assign a significantly higher number than regular policies that allow traffic.
+
+### How to
+
+Although you can use any of the following policies to create default deny for Kubernetes pods, we recommend using the Calico policy, non-namespaced.
+
+- [Create default deny-all ingress and egress Calico policy, non-namespaced](#create-default-deny-all-ingress-and-egress-calico-policy-non-namespaced)
+- [Create default deny-all ingress and egress Calico policy, namespaced](#create-default-deny-all-ingress-and-egress-calico-policy-namespaced)
+- [Create default deny all ingress and egress Kubernetes policy, namespaced](#create-default-deny-all-ingress-and-egress-Kubernetes-policy-namespaced)
+
+#### Create default deny-all ingress and egress Calico policy, non-namespaced
+
+In the following example, we specify a default deny global network policy that applies to workload endpoint resources in all namespaces, and host endpoint resources. The absence of rules means no traffice is allowed by the policy. 
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: default-deny
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+```
+
+#### Create default deny-all ingress and egress Calico policy, namespaced  
+
+In the following example, we specify a default deny Calico network policy for workloads in the namespace, **engineering**. Because Calico network policy is namespaced, the **order:20** is higher relative to policies in this namespace to ensure it remains in effect at all times. Note that the **selector: all()** is not required, but is used for clarity. 
+
+```
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: default-deny
+  namespace: engineering
+spec:
+  order: 20
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Deny
+  egress:
+  - action: Deny   
+```
+
+#### Create default deny-all ingress and egress Kubernetes policy, namespaced
+
+The following example is a Kubernetes default deny-all ingress and egress network policy. It prevents all traffic to/from all pods in the **policy-demo**namespace, and does not explicitly allow any traffic. Because the default changes when pods are selected by a network policy, the result is: **deny all ingress and egress traffic**. (Unless the traffic is allowed by another network policy).
```

Comment:
A general comment (for here, and the rest of this doc and others).

> The following example is a Kubernetes default deny-all ingress and egress network policy. It prevents all traffic to/from all pods in the **policy-demo**namespace, and does not explicitly allow any traffic. Because the default changes when pods are selected by a network policy, the result is: **deny all ingress and egress traffic**. (Unless the traffic is allowed by another network policy).

This is one big line, which makes it difficult to apply comments in the GitHub UI to specific sentences, and also makes it hard to read in a number of common editors that don't line-wrap.

We usually try to keep single lines to an absolute max of 120 characters, and then add a return after that.

Reply:
Good to know. I will be more mindful of this limitation in the future.